pkg/arvo/app/publish.hoon

```diff
@@ -542,6 +547,63 @@
 ::
 |_  bol=bowl:gall
 ::
+++  get-last-update
+  |=  [host=@p book-name=@tas]
+  ^-  (unit @da)
+  =/  book  (~(get by books) host book-name)
+  ?~  book  ~
+  =/  wen  date-created.u.book
+  %-  some
+  %-  ~(rep by notes.u.book)
+  |=  [[@tas =note] out=_wen]
+  ^-  @da
+  =?  out  (gth last-edit.note wen)
+    last-edit.note
+  =.  out
+    %-  ~(rep by comments.note)
+    |=  [[@da =comment] out=_out]
+    ?:  (gth date-created.comment out)
+      date-created.comment
+    out
+  out
+::
+++  get-notebook-from-date
+  |=  [host=@p book-name=@tas wen=@da]
+  ^-  notebook
+  =/  book  (~(got by books) host book-name)
+  %=  book
+      notes
+    %-  ~(rep by notes.book)
+    |=  [[nom=@tas not=note] out=(map @tas note)]
+    ^-  (map @tas note)
+    ?:  (gth last-edit.not wen)
+      (~(put by out) nom not)
+    =.  comments.not
+      %-  ~(rep by comments.not)
+      |=  [[nam=@da com=comment] out=(map @da comment)]
+      ?:  (gth date-created.com wen)
+        (~(put by out) nam com)
+      out
+    ?~  comments.not
+      out
+    (~(put by out) nom not)
+  ==
+::
+++  merge-notebooks
+  |=  [base=notebook diff=notebook]
+  ^-  notebook
+  %=  base
+      notes
```

Comment:
Don't we want to update things like title, description etc also? Those might have changed in our absence too.

So, maybe this should be `%=(diff ...)` instead.

Reply:
Good catch. I also just noticed this won't handle the case of deleted notes.